hal/src/nRF52840/ble_hal.cpp

```diff
@@ -15,739 +15,3147 @@
+            halProperties |= BLE_SIG_CHAR_PROP_BROADCAST;
+        }
+        if (properties.read) {
+            halProperties |= BLE_SIG_CHAR_PROP_READ;
+        }
+        if (properties.write_wo_resp) {
+            halProperties |= BLE_SIG_CHAR_PROP_WRITE_WO_RESP;
+        }
+        if (properties.write) {
+            halProperties |= BLE_SIG_CHAR_PROP_WRITE;
+        }
+        if (properties.notify){
+            halProperties |= BLE_SIG_CHAR_PROP_NOTIFY;
+        }
+        if (properties.indicate) {
+            halProperties |= BLE_SIG_CHAR_PROP_INDICATE;
+        }
+        if (properties.auth_signed_wr) {
+            halProperties |= BLE_SIG_CHAR_PROP_AUTH_SIGN_WRITES;
+        }
+        return halProperties;
+    }
+
+    static void toPlatformCharProps(uint8_t halProperties, ble_gatt_char_props_t* properties) {
+        memset(properties, 0x00, sizeof(ble_gatt_char_props_t));
+        properties->broadcast = (halProperties & BLE_SIG_CHAR_PROP_BROADCAST) ? 1 : 0;
+        properties->read = (halProperties & BLE_SIG_CHAR_PROP_READ) ? 1 : 0;
+        properties->write_wo_resp = (halProperties & BLE_SIG_CHAR_PROP_WRITE_WO_RESP) ? 1 : 0;
+        properties->write = (halProperties & BLE_SIG_CHAR_PROP_WRITE) ? 1 : 0;
+        properties->notify = (halProperties & BLE_SIG_CHAR_PROP_NOTIFY) ? 1 : 0;
+        properties->indicate = (halProperties & BLE_SIG_CHAR_PROP_INDICATE) ? 1 : 0;
+        properties->auth_signed_wr = (halProperties & BLE_SIG_CHAR_PROP_AUTH_SIGN_WRITES) ? 1 : 0;
+    }
+};
+
+uint16_t GattBase::desiredAttMtu_;
+Vector<GattBase::LinkAttMtu> GattBase::attMtuList_;
 
-// Advertising duration in 10 ms units (fast mode)
-const auto FAST_ADVERT_TIMEOUT = MSEC_TO_UNITS(180000, UNIT_10_MS); // Fast mode
-const auto SLOW_ADVERT_TIMEOUT = 0; // Advertise indefinitely (slow mode)
+class BleObject {
+public:
+    class BleEventDispatcher;
+    class BleGap;
+    class Broadcaster;
+    class Observer;
+    class ConnectionsManager;
+    class GattServer;
+    class GattClient;
 
-// Minimum acceptable connection interval in 1.25 ms units
-const auto MIN_CONN_INTERVAL = MSEC_TO_UNITS(30, UNIT_1_25_MS);
+    std::unique_ptr<BleEventDispatcher> dispatcher;         /**< BLE event dispatcher. */
+    std::unique_ptr<BleGap> gap;                            /**< BLE GAP instance. */
+    std::unique_ptr<Broadcaster> broadcaster;               /**< BLE Broadcaster instance. */
+    std::unique_ptr<Observer> observer;                     /**< BLE Observer instance. */
+    std::unique_ptr<ConnectionsManager> connectionsMgr;     /**< BLE connections manager instance. */
+    std::unique_ptr<GattServer> gatts;                      /**< BLE GATT server instance. */
+    std::unique_ptr<GattClient> gattc;                      /**< BLE GATT client instance. */
 
-// Maximum acceptable connection interval in 1.25 ms units
-const auto MAX_CONN_INTERVAL = MSEC_TO_UNITS(45, UNIT_1_25_MS);
+    static BleObject& getInstance(void);
```

Comment:
C++ code does not require unnamed `void` argument for functions/methods not taking any arguments. Let's remove all of such instances.